projects/plugins/jetpack/extensions/shared/register-jetpack-block.native.js

```diff
@@ -14,43 +19,87 @@ function requiresPlan( unavailableReason, details ) {
  */
-export default function registerJetpackBlock( name, settings, childBlocks = [] ) {
-	const { available, details, unavailableReason } = getJetpackExtensionAvailability( name );
+export default function registerJetpackBlock(
+	nameOrMetadata,
+	settings,
+	childBlocks = [],
+	prefix = true
+) {
+	const name = typeof nameOrMetadata === 'string' ? nameOrMetadata : nameOrMetadata.name;
+	const isNamePrefixed = name.startsWith( JETPACK_PREFIX );
+	const rawName = isNamePrefixed ? name.slice( JETPACK_PREFIX.length ) : name;
+
+	const { available, details, unavailableReason } = getJetpackExtensionAvailability( rawName );
 
 	const requiredPlan = requiresPlan( unavailableReason, details );
+	const jpPrefix = prefix || isNamePrefixed ? JETPACK_PREFIX : '';
 
 	if ( ! available && ! requiredPlan ) {
 		if ( 'production' !== process.env.NODE_ENV ) {
 			// eslint-disable-next-line no-console
 			console.warn(
-				`Block ${ name } couldn't be registered because it is unavailable (${ unavailableReason }).`
+				`Block ${ rawName } couldn't be registered because it is unavailable (${ unavailableReason }).`
 			);
 		}
 		return false;
 	}
 
-	const result = registerBlockType( `jetpack/${ name }`, {
+	const prefixedName = jpPrefix + rawName;
+	const result = registerBlockType( nameOrMetadata === 'object' ? nameOrMetadata : prefixedName, {
 		...settings,
-		title: betaExtensions.includes( name ) ? `${ settings.title } (beta)` : settings.title,
+		title: betaExtensions.includes( prefixedName ) ? `${ settings.title } (beta)` : settings.title,
 		edit: settings.edit,
 		example: requiredPlan ? undefined : settings.example,
 	} );
 
 	if ( 'production' !== process.env.NODE_ENV ) {
 		// eslint-disable-next-line no-console
-		console.log( `Block jetpack/${ name } registered.` );
+		console.log( `Block ${ prefixedName } registered.` );
 	}
 
 	// Register child blocks. Using `registerBlockType()` directly avoids availability checks -- if
 	// their parent is available, we register them all, without checking for their individual availability.
 	childBlocks.forEach( childBlock =>
-		registerBlockType( `jetpack/${ childBlock.name }`, childBlock.settings )
+		registerBlockType( jpPrefix + childBlock.name, childBlock.settings )
 	);
 
 	return result;
 }
+
+/**
+ * Wrapper around registerJetpackBlock to register a block by specifying its metadata.
+ *
+ * @param {object} metadata - Metadata of the block (content of block.json)
+ * @param {object} settings - See registerJetpackBlock.
+ * @param {object} childBlocks - See registerJetpackBlock.
+ * @param {boolean} prefix - See registerJetpackBlock.
+ * @returns {object|boolean} Either false if the block is not available, or the results of `registerBlockType`
+ */
+export function registerJetpackBlockFromMetadata( metadata, settings, childBlocks, prefix ) {
+	const mergedSettings = {
+		...settings,
+		icon: getBlockIconProp( metadata ),
+		title: metadata.title,
```

Comment:
**Discrepancy between native and web version:** I noticed that the web version is not adding `title` here. Following https://github.com/Automattic/jetpack/pull/33890/commits/f043df7738f5c7d28b4a2f6affd9e8812e314fdd, I assume we need it in the native version. Is this accurate?

https://github.com/Automattic/jetpack/blob/eebad7d58da476499ccc090d3b78c02547918c33/projects/plugins/jetpack/extensions/shared/register-jetpack-block.js#L81-L85

Reply:
@fluiddot, yes, I noticed that the title is only passed as part of the `metadata` object for native, and not with the `settings` object. I haven't been able to pinpoint the exact reason for the difference, however. 🤔

Reply:
That's interesting. I presume the blocks' title might be processed differently in the native version so we need to explicitly include it here. I don't foresee this as an important divergence between versions, so we can keep it as-is for now. In the future, if we want to unify the Jetpack blocks registration logic, we could consider revisiting this part.